app/code/Magento/Catalog/Model/ResourceModel/Category/Collection.php

```diff
@@ -402,6 +460,23 @@ public function addRootLevelFilter()
         return $this;
     }
 
+    /**
+     * Add navigation max depth filter
+     *
+     * @return $this
+     */
+    public function addNavigationMaxDepthFilter()
```

Comment:
Adding public methods to API classes in patch releases is prrrohibited by our backward compatibility policy